src/tasks/mod.rs

```diff
@@ -514,27 +494,47 @@ pub async fn add_consensus_task(
 /// add the Data Availability task
 /// # Panics
 /// Is unable to panic. This section here is just to satisfy clippy
-pub async fn add_da_task(
+pub async fn add_da_task<
+    TYPES: NodeType<ConsensusType = SequencingConsensus, Time = ViewNumber>,
+    I: NodeImplementation<
+        TYPES,
+        Leaf = SequencingLeaf<TYPES>,
+        ConsensusMessage = SequencingMessage<TYPES, I>,
+    >,
+>(
     task_runner: TaskRunner,
-    event_stream: ChannelStream<GlobalEvent>,
-) -> TaskRunner {
+    event_stream: ChannelStream<SequencingHotShotEvent<TYPES, I>>,
+    committee_exchange: CommitteeEx<TYPES, I>,
+) -> TaskRunner
+where
+    I::Exchanges: SequencingExchangesType<TYPES, Message<TYPES, I>>,
+    CommitteeEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Certificate = DACertificate<TYPES>,
+        Commitment = TYPES::BlockType,
+    >,
+{
     // build the da task
-    let da_state = DATaskState {};
+    let da_state = DATaskState {
+        cur_view: TYPES::Time::new(0),
+        high_qc: QuorumCertificate::<TYPES, I::Leaf>::genesis(),
+        committee_exchange: committee_exchange.into(),
+        vote_collector: (TYPES::Time::new(0), async_spawn(async move {})),
```

Comment:
I ended up storing the task_id rather than a future/JoinHandle here.  The GlobalRegistry allows you to manually shut down, so that's what I use.